frontend/src/metabase/collections/components/CreateCollectionForm/tests/setup.tsx

```diff
@@ -0,0 +1,54 @@
+/* istanbul ignore file */
+import { TokenFeatures, User } from "metabase-types/api";
+import {
+  createMockCollection,
+  createMockTokenFeatures,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import { createMockState } from "metabase-types/store/mocks";
+import { setupEnterprisePlugins } from "__support__/enterprise";
+import { setupCollectionsEndpoints } from "__support__/server-mocks";
+import { mockSettings } from "__support__/settings";
+import { createMockEntitiesState } from "__support__/store";
+import { renderWithProviders } from "__support__/ui";
+import CreateCollectionForm from "../CreateCollectionForm";
+
+const ROOT_COLLECTION = createMockCollection({
+  id: "root",
+  name: "Our analytics",
+  can_write: true,
+});
+
+export interface SetupOpts {
+  user?: User;
+  tokenFeatures?: TokenFeatures;
+  hasEnterprisePlugins?: boolean;
+}
+
+export const setup = ({
+  user = createMockUser({ is_superuser: true }),
+  tokenFeatures = createMockTokenFeatures(),
+  hasEnterprisePlugins = false,
+}: SetupOpts = {}) => {
+  const settings = mockSettings({ "token-features": tokenFeatures });
+  const onCancel = jest.fn();
+
+  if (hasEnterprisePlugins) {
+    setupEnterprisePlugins();
+  }
+  setupCollectionsEndpoints({
+    collections: [],
+    rootCollection: ROOT_COLLECTION,
+  });
+  renderWithProviders(<CreateCollectionForm onCancel={onCancel} />, {
```

Comment:
With this new setup, we limit ourselves from testing when `onCancel` is null. I still think we should test that. e.g. this test is gone
https://github.com/metabase/metabase/blob/bfbfb7abf3f7a6e353adfee26a350b29ee3caee5/frontend/src/metabase/collections/containers/CreateCollectionForm.unit.spec.tsx#L70-L75